src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowHotfixFinishMojo.java

```diff
@@ -154,6 +154,31 @@ public void execute() throws MojoExecutionException, MojoFailureException {
                     throw new MojoFailureException("Hotfix branch with name '" + branch + "' doesn't exist. Cannot finish hotfix.");
                 }
                 hotfixBranchName = branch;
+            } else if (StringUtils.isBlank(hotfixVersion)) {
```

Comment:
**DISCLAIMER:** I'm not a maintainer, so feel free to ignore what I'm saying. 😉

This is basically the same block as in [`GitFlowReleaseFinishMojo`](https://github.com/aleksandr-m/gitflow-maven-plugin/blob/master/src/main/java/com/amashchenko/maven/plugin/gitflow/GitFlowReleaseFinishMojo.java#L191-L218).

I wonder if it makes sense / is possible to avoid the duplicated code by, e.g., having an intermediate layer like `AbstractGitFlowFinishMojo`, which encloses this functionality. (Although it looks like `GitFlowFeatureFinishMojo` couldn't benefit from it.)

Reply:
Abstraction via inheritance is an obvious improvement since you write less code, right? But in many cases inheritance becomes much more inconvenient to maintain, than just copy some code, that has potential to diverge a little in the future.

If any, I would prefer using composition over inheritance everywhere. At least you can interchange the solution with a very less effort.

Reply:
>Abstraction via inheritance is an obvious improvement since you write less code, right?

No, but thanks for ELI5. 😉